edb/server/compiler/compiler.py

```diff
@@ -1557,6 +1559,72 @@ def _compile_ql_query(
     is_explain = explain_data is not None
     current_tx = ctx.state.current_tx()
 
+    # Embed the normalized EdgeQL query info in the SQL as a comment.
+    # It has 2 layers of embedded JSON:
+    #   1. sql_info: the outer comment consumed by the edb_stat_statements
+    #      extension, used to replace the compiled SQL with the frontend query
+    #      in this JSON, as well as the `queryId` (unique index of statistics).
+    #   2. query_info: the inner comment that is stored in pg_stat_statements,
+    #      used as a supplement storage of the additional query information,
+    #      like query_type, output_format, etc.
+    sql_info: Dict[str, Any] = {}
+    if (
+        not ctx.bootstrap_mode
+        and ctx.backend_runtime_params.has_stat_statements
+    ):
```

Comment:
Do we want whether we collect this to also depend on a config flag? This seems like a lot of extra stuff to store unconditionally.
(Curious about @elprans's opinion on this)

Reply:
Answered in the meeting: we do need a config to turn the tracking on/off.

Follow-up question: do we want to make this config `affects_compilation` (thus controlling the compiled SQL with or without the magic JSON-comment prefix, but impacting query cache), or a runtime session config that the extension can read and behave (but we end up with prefixed SQLs all the time)?